SonarQube.Client/Requests/PagedRequestBase.cs

```diff
@@ -73,10 +73,12 @@ public override async Task<TResponseItem[]> InvokeAsync(HttpClient httpClient, C
                 return allResponseItems.ToArray();
             }
 
-            Logger.Warning("Sonar web maximum API response limit reached.");
+            if (allResponseItems.Count == MaximumItemsCount)
```

Comment:
We should probably validate that caller hasn't set the `ItemsLimit` to a number greater than `MaximumItemsCount`, but that's a separate problem.

Reply:
It's also difficult to write a test for without refactoring, so I'll add an assertion for now.